### ISSUE TITLE
Vscode doesn't show change in opened file when having same file name with different casing

### ISSUE THREAD

**Issue by rquispeq** [microsoft/vscode#107738]
Issue type: **Bug**
When I change a file, the section where shows me the changes in git doesn't shows up.

This happen in some files, the only explanation to me its because the file has an issue with git for having 2 names, uppercase and lowercase:

There are these screenshots:

![image](https://user-images.githubusercontent.com/23042251/94603450-e60fb280-025b-11eb-855b-b13183e56d9a.png)

You can see vscode knows that the file has changes in the source control tab, but doesn't shows in the opened file.

When I press ctrl + P and search the file, I find 2 files with the first letter in uppercase and lowercase, but its the same file. I think this is the problem with git but somehow affect vscode too.

![image](https://user-images.githubusercontent.com/23042251/94603714-3e46b480-025c-11eb-8efb-4b19b8fc56dd.png)

And when I click the file in source control tab, It shows me the changes I made. In this case, I move all the code one space but added a line too.
![image](https://user-images.githubusercontent.com/23042251/94604195-eeb4b880-025c-11eb-962a-81d555139174.png)

I hope if someone knows how to fix this. And but the way sorry for my english.

**Comment by joaomoreno** [comment]
I assume you're on Linux?

**Comment by rquispeq** [comment]
Hi, I'm on windows 10, the gutter that you mentioned it doesn't work for both files. These filenames are the same file by the way.

![image](https://user-images.githubusercontent.com/23042251/94740273-fb5a0f00-0337-11eb-9923-ccfd88bea05a.png)

![image](https://user-images.githubusercontent.com/23042251/94740298-06ad3a80-0338-11eb-9764-bb0bcd759d21.png)

The variable $test is the new line that I added.

**Comment by joaomoreno** [comment]
And it only happens on such files? All other files are OK?

**Comment by rquispeq** [comment]
Other files are ok, I have this problem on files with the uppercarse/lowercase cases. 

I open the entire project on another texteditor and when I search the same file, it shows me just one match, the filename in lowercase.

**Comment by joaomoreno** [comment]
> I open the entire project on another texteditor and when I search the same file, it shows me just one match, the filename in lowercase.

Yeah it's a pretty bad idea to have such files. Many tools will break with them.

**Comment by tomkra** [comment]
I can confirm this issue on Ubuntu 20.04, vscode version 1.49.3, all extensions disabled. I've firstly encountered problem after upgrade to latest version from 1.49.2. Source control changes are detected but indication by color in editor and in file explorer is missing.

It's interesting that for me this is happening only in projects, that have git origin on self hosted domain git domain. On projects where origin is github.com or gitlab.com It's working fine. Could this be somehow related?

Check attached image but in my project I have only one file `meet_subclass.rb` with no upper/lowercase cases.

![issue](https://user-images.githubusercontent.com/9742214/95241683-66ab5180-080e-11eb-9ed1-8044e64e1cd4.png)

**Comment by joaomoreno** [comment]
@tomkra Your issue seems to be related to symlinks, not filename casing: https://github.com/microsoft/vscode/issues/5970

**Comment by msftrncs** [comment]
Adding to the filename casing issue: (Windows)

My project file has the path `\\server\share\path` and GIT shows the path as `\\SERVER\Share\path` and the actual naming is `\\SERVER\share\path`.   I do not know where the version of the path is coming from that the GIT/SCM is coming up with.   Most workspaces are okay, the case matches in both VS Code and in GIT, even though its not the actual case of the root path.

The loss of the change indicators is recent.  Before I also used to get the same repository listed twice (different case for each path, and I noticed some files where open with different cases, from the main work folder), so it seems there was some recent change to either GIT or VSCode that improved handling of differently cased repositories that are actually the same, but has broken the change indicators.

If I carefully open my project workspace using the case that matches what GIT seems to be using, then the change indicators work again, but since the case GIT is using is not even the real case from the file system, this is not a preferred workaround.

**Comment by msftrncs** [comment]
I thought this was entirely about case, but I am not so sure.   I carefully opened by workspace (which has a relative workfolder) so as the case would match what GIT seems to be reporting, and only some files work.   I can tell the files that work, because the GIT log reports `git ls-files --stage -- _filename with full path_` upon switching tabs, etc.   There is no reaction in the GIT log to files that don't work, when they are opened or switched back to their editor tab.  I am not seeing any rhyme for the files that work vs the ones that do not.   The SCM view indicates they are all modified and the File View indicates the same, but the change indicators only seem to work on certain files.

**Comment by msftrncs** [comment]
I finally had to close all persistently open files in the workspace, close the workspace, and then reopen the workspace file with the case specifically matching what GIT keeps reporting in the SCM for the repository in question, and now each file I open shows the correct change indicators.  I did try opening the workspace with a different path casing with no files open, but that didn't affect what the SCM view showed for a path.   I have also opened this particular repository separately using various casing, and the SCM always reports the path with a specific case. 

This particular workspace has multiple repositories (folders) included.   The other repositories are not on the same server, but they were working fine, because the case of both the File view and the SCM view were matching the whole time.

Not knowing GIT very well, I finally determined that the specific path issue can be verified by `git rev-parse --absolute-git-dir`.   I still don't understand how it arrives at this name, but it seems to be consistent at least on the same server.  This is a new server (files were moved to this new server which then assumed the name of the old server) and that is also a recent change.  Its a Windows Server 2019 based file server.   I tested the same command on other (older) servers that I have repositories I can access, and there the result followed the current CASE of the working folder.
